test/unit/oidc_jwt_token_validation_test.rb

```diff
@@ -0,0 +1,76 @@
+require 'test_helper'
+require 'ostruct'
+
+class OidcJwtTokenValidationTest < ActiveSupport::TestCase
+  # Jwt token created with the folloing payload:
+  # { "name": "jwt token", "iat": 1557224758, "exp": 1557228982, "typ": "Bearer", "aud": "rest-client"}
+  let(:token) { 'eyJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoxMjMsImlhdCI6MTUxNDgwNDQwMCwianRpIjoiM2U4Mjg2OTQwZWIxNjJlYzczNWY4YTVhYWM1OTI2MDM3ZmRjN2YwNWU1MjFhNzQyMzFhNjVmMjU1N2E4YWY5NCJ9.TSM2xMnMuMEWwAVoIidyLIwJElJQZVQvcfaxyVA7cDI' }
+  let(:signature) { 'TSM2xMnMuMEWwAVoIidyLIwJElJQZVQvcfaxyVA7cDI' }
+
+  test 'if all checks pass for valid_jwt?' do
+    OidcJwtTokenValidate.any_instance.expects(:valid_signature?).returns(true)
+    OidcJwtTokenValidate.any_instance.expects(:type_bearer?).returns(true)
```

Comment:
I think all of these tests should not stub these methods but rather test them. It would be better to feed them with actual data that we can use.
Bear in mind that tests won't work when the jwt token expires. To keep this future proof rails has luckily [invented time traveling](https://api.rubyonrails.org/v5.2.3/classes/ActiveSupport/Testing/TimeHelpers.html).

Reply:
Yes, I will add tests for cases where the token expires. Thanks for the link :)